shared/src/main/scala/fr/hmil/roshttp/SimpleHttpResponse.scala

```diff
@@ -0,0 +1,59 @@
+package fr.hmil.roshttp
+
+import java.nio.ByteBuffer
+import java.util.concurrent.TimeUnit
+
+import monifu.reactive.Observable
+
+import monifu.concurrent.Implicits.globalScheduler
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.duration.FiniteDuration
+
+/**
+ * An HTTP response obtained via an [[HttpRequest]]
+ */
+class SimpleHttpResponse(
+    val statusCode: Int,
+    val body: String,
+    val headers: HeaderMap[String])
+  extends HttpResponse
+
+object SimpleHttpResponse extends HttpResponseFactory[SimpleHttpResponse] {
+  override def apply(
+      statusCode: Int,
+      bodyStream: Observable[ByteBuffer],
+      headers: HeaderMap[String])
+      (implicit ec: ExecutionContext, config: HttpConfig): Future[SimpleHttpResponse] = {
+
+    val charset = HttpUtils.charsetFromContentType(headers.getOrElse("content-type", null))
+
+    bodyStream
+      .buffer(FiniteDuration(config.bodyCollectTimeout, TimeUnit.SECONDS))
+      .map({ seq =>
+        // Allocate maximum expected body length
+        val buffer = ByteBuffer.allocate(seq.length * config.streamChunkSize)
+        val totalBytes = seq.foldLeft(0)({(count, chunk) =>
+          buffer.put(chunk)
+          count + chunk.limit
+        })
+        buffer.limit(totalBytes)
+        getStringFromBuffer(buffer, charset)
+      })
+      // TODO: We might not want to depend on monifu's Scheduler
+      .asFuture
```

Comment:
Well, you can't have your cake and eat it too.

If you're basing yourself on top of Monix, you have to play by Monix' rules. If it wants an implicit `Scheduler`, then yes, you need to demand a `Scheduler` from your users. Unless you can internally synthesize a valid `Scheduler` from an `ExecutionContext`, but I doubt that (otherwise Monix would take ECs to begin with).

If you don't want to impose the Scheduler tax on your users (which, I agree, is desirable), then you cannot use parts of the Monix API that require Schedulers.